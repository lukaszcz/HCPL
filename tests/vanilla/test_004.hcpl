
let fib = \&n . {
   if == n 0 then 0
   else if == n 1 then 1
   else { + (fib (- n 1)) (fib (- n 2)) } };

let fibe = \!n .
   if == n 0 then 0
   else if == n 1 then 1
   else + (fibe (- n 1)) (fibe (- n 2));

print (fibe 5);
print (fibe 12);

print ((\&x . x) 5);
print ((\&x . + x 3) 5);

print (fib 5);
print (fib 12);
