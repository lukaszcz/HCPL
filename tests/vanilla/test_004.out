5
144
5
8
5
144
